src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt

```diff
@@ -44,16 +59,35 @@ interface VisitRepository : JpaRepository<Visit, Long>, JpaSpecificationExecutor
       "v.visitStart < :endDateTime and " +
       "v.visitRoom = :visitRoom and " +
       "(v.visitRestriction = 'OPEN' or v.visitRestriction = 'CLOSED') and " +
-      " (v.visitStatus = 'BOOKED' or v.visitStatus = 'RESERVED') " +
+      "v.visitStatus = 'BOOKED'  " +
       "group by v.visitRestriction"
   )
-  fun getCountOfActiveSessionVisitsForOpenOrClosedRestriction(
+  fun getCountOfBookedSessionVisitsForOpenOrClosedRestriction(
     prisonId: String,
     visitRoom: String,
     startDateTime: LocalDateTime,
     endDateTime: LocalDateTime
   ): List<VisitRestrictionStats>
 
+  @Query(
+    "SELECT v.visit_restriction as visitRestriction, count(*) as count  FROM visit v " +
+      "WHERE v.prison_id = :prisonId and " +
+      "v.visit_start >= :startDateTime and " +
+      "v.visit_start < :endDateTime and " +
+      "v.visit_room = :visitRoom and " +
+      "v.visit_restriction in ('OPEN','CLOSED') and " +
+      "v.visit_status = 'RESERVED' and v.modify_timestamp >= NOW() - (make_interval(mins => :expiredPeriodMinutes))" +
+      "group by v.visit_restriction",
+    nativeQuery = true
+  )
+  fun getCountOfReservedSessionVisitsForOpenOrClosedRestriction(
+    prisonId: String,
+    visitRoom: String,
+    startDateTime: LocalDateTime,
+    endDateTime: LocalDateTime,
+    expiredPeriodMinutes: Int
```

Comment:
Same as the earlier comment, will it be better to pass the expiredPeriod as LocalDateTime instead of int post calculation in Java code.

Reply:
commented on this issue above